stdlib/Printf/src/Printf.jl

```diff
@@ -3,25 +3,306 @@
 __precompile__(true)
 
 module Printf
-# the macro implementations here exactly mirrors the
-# macros left in base/printf.jl, and uses the utility there
+# the implementations here is copied from
+# what is left in base/printf.jl, and uses the utility there
```

Comment:
It would be better not to copy the code. If we're going to use the macros plus `eval`, why not wrap calls to the existing macros in function expressions and `eval` those?

Reply:
The code was copied, then modified.
In the old version, there were only the macro definitions, and they were exact copies.
Actually I only do not understand, why base/printf.jl is still there. Maybe a bootstrap problem.

Reply:
The format-string parser had to be modified to support the new features. I tried to copy-and-modify as little as possible. So I call back to the code-generator functions `gen_d`, `gen_f` etc. and they generate code, which uses other functions, which are in `base/printf.jl`.
For the additional feature of variable width and precision specifiers like in `"%*.*f"` I would need to modify the `gen_x` functions also.